service/push_notification.py

```diff
@@ -43,31 +45,32 @@ def update_firebase_token(self, calendar_id, firebase_token):
                 calendar['firebase_token'] = firebase_token
                 db.hmset(key, calendar)
         return "OK"
-
+    @celery.task(name='push_notification.refresh', bind=True)
     def refresh(self):
-        rooms_query = (
-            {"query": "{ allRooms { rooms { calendarId, firebaseToken } } }"})
-        headers = {'Authorization': 'Bearer %s' % api_token}
-        all_rooms = requests.post(url=url, json=rooms_query, headers=headers)
+        with app.test_request_context():
+            rooms_query = (
+                {"query": "{ allRooms { rooms { calendarId, firebaseToken } } }"})
+            headers = {'Authorization': 'Bearer %s' % api_token}
+            all_rooms = requests.post(url=url, json=rooms_query, headers=headers)
 
-        rooms = all_rooms.json()['data']['allRooms']['rooms']
-        selected_calendar = {}
-        selected_calendar_key = ''
-        for room in rooms:
-            for key in db.keys('*Calendar*'):
-                calendar = db.hgetall(key)
-                if calendar['calendar_id'] == room['calendarId']:
-                    selected_calendar = calendar
-                    selected_calendar_key = key
-                    break
-            update_calendar(selected_calendar, selected_calendar_key, room)
+            rooms = all_rooms.json()['data']['allRooms']['rooms']
+            selected_calendar = {}
+            selected_calendar_key = ''
+            for room in rooms:
+                for key in db.keys('*Calendar*'):
+                    calendar = db.hgetall(key)
+                    if calendar['calendar_id'] == room['calendarId']:
+                        selected_calendar = calendar
+                        selected_calendar_key = key
+                        break
+                update_calendar(selected_calendar, selected_calendar_key, room)
 
-        data = {
-            "message": "Calendars saved successfully"
-        }
-        response = jsonify(data)
+            data = {
+                "message": "Calendars saved successfully"
+            }
+            response = jsonify(data)
 
-        return response
+            return "Calendars saved successfully"
```

Comment:
@michael-basweti I guess you want to return `response` here instead of the `String`.

Reply:
If you look at the response, it's basically a string, I decided to go with the string since we when using celery, we are not returning a json data. If you look at the code, response is in json.